### PR TITLE
drop deprecated random-bytestring dependency

### PR DESCRIPTION
**Pull request by larskuhtz**

### CODE REVIEW

File: src/Chainweb/Utils.hs [hunk 1]
```diff
@@ -927,6 +930,30 @@ leadingZeros b =
...
+-- and life locks.
+--
+-- However, use of the generator is still sequentialized. Thus, for long
+-- 'ByteString's it can be more efficient to split the generator to speed up
+-- concurrent access.
+
+randomShortByteString :: MonadIO m => Natural -> m BS.ShortByteString
+randomShortByteString n
+    -- don't split the generators for less than 64 words.
+    | n < 8 * 64 = getStdRandom $ genShortByteString (int n)
```
Comment (emilypi) [inline src/Chainweb/Utils.hs:948]: May as well just inline this and save a calculation. 

```suggestion
    -- 512 = 8 * 64
    | n < 512 = getStdRandom $ genShortByteString (int n)
```

File: src/Chainweb/Utils.hs [hunk 2]
```diff
@@ -927,6 +930,30 @@ leadingZeros b =
...
+    | otherwise = fst . genShortByteString (int n) <$> newStdGen
+
+randomByteString :: MonadIO m => Natural -> m B.ByteString
+randomByteString n
+    -- don't split the generators for less than 64 words.
+    | n < 8 * 64 = getStdRandom $ genByteString (int n)
```
Comment (emilypi) [inline src/Chainweb/Utils.hs:954]:
```suggestion
    -- 512 = 8 * 64
    | n < 512 = getStdRandom $ genByteString (int n)
```

**Review by emilypi — APPROVED** [review]
Optional suggestions. Otherwise 👍

File: src/Chainweb/Utils.hs [hunk 3]
```diff
@@ -927,6 +930,30 @@ leadingZeros b =
     in out
 {-# INLINE leadingZeros #-}
 
+-- -------------------------------------------------------------------------- --
+-- Random ByteString
+--
+-- 'getStdRandom' provides an generator that is stored in an 'IORef' and updated
+-- via an optimistic atomci swap. 'atomicModifyIORef'' is implemented such that
+-- the swapped pointer is updated lazily, which minimizes the chance of retries
+-- and life locks.
```
Comment (larskuhtz) [inline src/Chainweb/Utils.hs:939]:
```suggestion
-- 'getStdRandom' provides a generator that is stored in an 'IORef' and updated
-- via an optimistic atomic swap. 'atomicModifyIORef'' is implemented such that
-- the swapped pointer is updated lazily, which minimizes the chance of retries
-- and life locks.
```

**Pull request merged**